### ISSUE TITLE
Utmost necessity of adding quaternionic intervals

### ISSUE THREAD

**Issue by arseniiv** [xenharmonic-devs/sonic-weave#202]
As we all know, quaternions allow one to seamlessly rotate their point of view both in 3D space and even in time (by using biquaternions, or equivalently, multiplying by two different quaternions from left and right), so it would be very useful for working with scales and temperaments.

For example, the famous case of a true tritone `[1/2>`. It’s debated to be both a kind of fifth (`d5`) and fourth (`Â4`) and also, sometimes, a kind of octave (`dddddd8`). Quaternions would allow fluidity in changing one’s understanding and acceptance of all of these options, with uncountable many others.

I don’t know how precisely it should be done and if JS has sufficient computing power but at I made a couple commits with tentative implementation of quaternions. Also I changed every other line of code because it wasn’t working with my implementation.

Unfortunately I then deleted my whole local repo. But hopefully git has ways to circumvent that and restore lost commits? Thus, this issue. (I’m very much confident the PR would be automatically accepted other than that I can’t make it.)

**Comment by frostburn** [comment]
This is the single most important discovery in the history of music!
I pray that you will be able to recover these lost commits. :pray: The future of all musicians depends on it! :musical_score: :milky_way: 

Once the PR is merged you will be knighted by the king-and-queen :crown: of microtones and will henceforth be known as sær arseniiv, the savior of xen, champion of spoob and all-around swell person to be on the internet with.

**Comment by arseniiv** [comment]
I can’t find relevant git documentation.

Seems the files were on a drive that’s very busy and they probably already rewritten by newer files being defragmented all over the place by maintenance services.

😭

**Comment by frostburn** [comment]
Devastating! We must assemble a team of elite xen hackers and hardware specialists that can retrieve as much of the lost data as possible!

**Comment by frostburn** [comment]
Any final words before I archive this issue? Your kind of genius insight is bound to be unique to the history of mankind and we shouldn't leave utterly difficult issues like this open to lure in unsuspecting programmers and doom them to eternal frustration.

**Comment by akselai** [comment]
I want to supplement a few words on this issue:
I propose an octonionic extension to the interval naming problem. As we are recently concerned with issues of division of various pythagorean intervals into eight parts (i.e. how eight is the maximal equal divisions one could do to the apotome in Pakkanen-extended FJS), and in light of how 8 is the maximum number of dimensions for a real division algebra, I think octonions will do an immensely great job of simplifying the interval naming issue. These octonionic cells give rise to important harmonic structures called octacells, named by A. Kaksli in her seminal PhD thesis [1]. See the image attached for a sense of what's to come.

This will greatly benefit the creation of the
![image-38](https://github.com/xenharmonic-devs/sonic-weave/assets/90088202/f5f036b9-bb12-4e3e-9445-1e11c2767860)
 future naming scheme of sonic-weave, known as XXXØHESHfIRHMÂP°WDMKWTEINDdFuJuSS (extended extended extended octonionic
Helmholtz-Ellis-Sabat-Hearne flavored Inthar-Richie-Hunt-Monzo Akselai-Pakkanen-0.5° Weigel-diamond-MOS Kees-Weil-Tenney-Euclid interordinal neutral dyadic dysfunctional unjust unsystematic system). 

[1] Über der Octonionische Urmutterakkord (On the Octonionic Grandmother Chord), Austrian Journal of Algebraic Psychoacoustics.

**Comment by arseniiv** [comment]
Yes I think what @akselai said supplants what I wished to add here (“it wasn’t me it was solely my cat”).
